### DISCUSSION TITLE
Criteria for Using Cached vs. Latest Metadata with metadata-cache.enabled in Iceberg on Trino

### DISCUSSION THREAD

**Discussion by byunks** [trinodb/trino#24100] [Q&A]
When using Iceberg's metadata-cache.enabled option set to true, I am curious about the criteria Trino uses to decide whether to use cached metadata or reload the latest metadata during query execution.

**Comment by hashhar** [comment]
please read https://github.com/trinodb/trino/discussions/23575

In iceberg current metadata version is stored in HMS (or whatever Iceberg Catalog you use) so it's always known to the engine whether new metadata needs to be read or the one in cache is up to date.
